drivers/unix/os_unix.cpp

```diff
@@ -179,8 +180,17 @@ Vector<String> OS_Unix::get_video_adapter_driver_info() const {
 }
 
 String OS_Unix::get_stdin_string() {
-	char buff[1024];
-	return String::utf8(fgets(buff, 1024, stdin));
+    String result;
+    ssize_t size = 0;
+    char buff[BUFSIZ];
+    int flags = fcntl(STDIN_FILENO, F_GETFL, 0);
+    if (flags != -1) {
+        while ((size = read(fcntl(STDIN_FILENO, F_SETFL, flags | O_NONBLOCK), buff, BUFSIZ))) {
+            buff[size] = '\0';
+            result += String::utf8(buff, size);
```

Comment:
Actually, this might be an issue if buffer cuts in the middle of a Unicode sequence, so it probably should read to the resizable `char` buffer and convert to `String` in the single call at the end.